### PR TITLE
[5.2] Complement existing 'get' method of 'Fluent' with 'set' method

### LINKED ISSUE #13481: Add explicit set(attribute, value = true) method to Fluent
To add an attribute to ``Illuminate\Support\Fluent`` we have to use magic methods. Contemporary IDEs don't like that. I would rather write ``.add('nullable').add('index')`` than ``.nullable().index()``, because it is more explicit (I am more careful with strings than with method calls because in the latter case I have IDE/runtime that can tell me I made a typo).

It will also make the API more consistent and predictable, because ``Fluent`` already has an explicit ``get`` method. The same with ``Request``—I can either use magic methods or ``$request->input('name')``, which I prefer.

### PR DESCRIPTION
**Pull request by ghost**
Complement existing `get` method of `Fluent` with `set` method. This allows writing explicit `set('nullable').set('index')` instead of magic `.nullable().index()`. This allows strict type checking in PhpStorm et al. instead of warnings. Fixes #13481 and #8675, relates to #8506.

### CODE REVIEW

File: src/Illuminate/Support/Fluent.php [hunk 1]
```diff
@@ -46,6 +46,20 @@ public function get($key, $default = null)
     }
 
     /**
+     * Explicilty set an attribute.
+     *
+     * @param  string  $key
+     * @param  mixed   $value
+     * @return $this
+     */
+    public function set($key, $value = true)
```
Comment (crynobone) [inline src/Illuminate/Support/Fluent.php]: `true` is a bad default for setting a value (need to follow `Illuminate\Config\Repository`). It should be either required or `null`

Again, it should be using an extended class under the `Schema` namespace as even agreed by @taylorotwell

Comment (crynobone) [inline src/Illuminate/Support/Fluent.php]: https://github.com/laravel/framework/pull/14520#issuecomment-235897408

**Comment by crynobone** [comment]
```php
<?php

namespace Illuminate\Database\Schema;

use Illuminate\Support\Fluent as BaseFluent;

class Fluent extends BaseFluent
{
    public function nullable()
    {
        $this->attributes['nullable'] = true;

        return $this;
    }

    // ...
}
```

File: src/Illuminate/Support/Fluent.php (hunk 1 above)
Comment (danilcha) [inline src/Illuminate/Support/Fluent.php]: About `true` — it is the default in `__call`:
```php
     public function __call($method, $parameters)
     {
         $this->attributes[$method] = count($parameters) > 0 ? $parameters[0] : true;
```

I agree, a specific class for migrations would be wonderful, but this change is orthogonal to it (maybe it will help in other use cases of `Fluent`). Moreover, in that comment @taylorotwell said that he is not sure if he wants to go that way.

**Comment by taylorotwell** [comment]
Yes, I think if we want to solve the IDE problem we just need an extension of Fluent specifically for Schema.
